### PR TITLE
feat(storage-proofs): use disk-trees feature as default

### PR DESCRIPTION
**Pull request by schomatis**
Alternatively we could invert the feature and have a `ram-trees`/`mem-trees` feature.

* [x] Adapt `disk-tree` test accordingly.

* [x] Update readme, we don't need to set this feature. Also, the RSS will be now 1x sector size.

Closes https://github.com/filecoin-project/rust-fil-proofs/issues/831.

-------------------------------

Benchmark:

```
RUSTFLAGS="-C target-cpu=native" cargo build --bin benchy --release

RUST_LOG=info /usr/bin/time -v ./target/release/benchy zigzag --size 262144
```
```
MASTER (disk-trees *not* set):
	User time (seconds): 10724.92
	System time (seconds): 12.24
	Percent of CPU this job got: 643%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 27:48.98
	Maximum resident set size (kbytes): 6693416

PR (disk-trees set as default):
	User time (seconds): 10712.94
	System time (seconds): 20.55
	Percent of CPU this job got: 648%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 27:34.49
	Maximum resident set size (kbytes): 926748
```

### CODE REVIEW

**Comment by dignifiedquire** [comment]
do you have a before benchmark, with mem trees?

File: storage-proofs/Cargo.toml [hunk 1]
```diff
@@ -48,7 +48,7 @@ log = "0.4.7"
 pretty_env_logger = "0.3.0"
 
 [features]
-default = []
+default = ["disk-trees"]
 simd = []
 asm = ["sha2/sha2-asm"]
 disk-trees = []
```
Comment (dignifiedquire) [inline storage-proofs/Cargo.toml]: `fil-proofs-tooling` and `filecoin-proofs` should get this feature too, which just proxies this, so there is control at that level

Comment (dignifiedquire) [reply storage-proofs/Cargo.toml]: actually fine to not do this for now, if we switch this on by default

**Comment by schomatis** [comment]
With https://github.com/filecoin-project/rust-fil-proofs/pull/837 merged this is now the final PR in the disk-MTs queue, will be resolved this week.

File: README.md [hunk 2]
```diff
@@ -159,24 +159,23 @@ To check that it's working you can inspect the replication log to find `using pa
...
 
-### Memory
-
-We try to generate the MTs in parallel to speed up the process but that takes 2 sector sizes per each layer (e.g., a 1 GiB sector may require, in the worst case scenario, up to 20 GiB of memory to hold the MTs alone). To reduce that (at the cost of speed) we have the (experimental) `disk-trees` feature to offload the MTs to disk when we don't use them. For example, to run the `zigzag` example with this feature you'd need to indicate so to `cargo`,
+In the most extreme case, to reduce time at the cost of *a lot* of memory consumption you can turn off the feature that stores MTs on disk (`disk-trees`) to generate them all on RAM and avoid disk I/O (if the HW doesn't have enough RAM to handle the MTs, roughly 20x the sector size, this won't have the desired effect as the OS will start backing them on disk anyway). For example, to run the `zigzag` example with this feature turned off you'd need to indicate so to `cargo`,
```
Comment (dignifiedquire) [inline README.md]: I don't think this is really true, I am not seeing any speed differences between mmemap and disk versions

Comment (schomatis) [reply README.md]: Me neither, but this might also depend on the HW, "the most extreme case" aimed at conveying something like that, could be rephrased.

Comment (schomatis) [reply README.md]: Now that you mention it, actually the mmap-MT should be replaced with the vector-MT (which would be selected only if the disk-trees is explicitly turned off).

**Comment by schomatis** [comment]
@dignifiedquire As there's no easy way to disable a default feature I think it would be better to have an optional `mem-trees` feature instead, that would switch from the (now) default disk MT to a vector based one when set.

**Comment by dignifiedquire** [comment]
@schomatis I agree switching to mem-tree feature and removing disk-tree feature makes more sense

**Pull request merged**